### ISSUE TITLE
Support more powerful regex ignore

### ISSUE THREAD

**Issue by innerlee** [bartosz-antosik/vscode-spellright#64]
In LaTeX, one want to ignore everything in an equation block. e.g.,
```latex
\begin{equation}
    |\Fix~\phi_H | \ge \sum_i b^i(M;\real).
\end{equation}
```
It would be great if we can achive this by adding the following pattern into the regex ignore list (currently not working)
```json
"/begin\\\\{equation\\\\}[\\\\s\\\\S]*?end\\\\{equation\\\\}/g"
```

**Comment by bartosz-antosik** [comment]
If I would provide you, within few days, with a test version of such a change in a VSIX package, would you agree to carry some tests? Maybe not only on LaTeX but Markdown/Plaintext documents as well?

**Comment by innerlee** [comment]
Sure! That's great 😊

**Comment by bartosz-antosik** [comment]
This is a link to VSIX of version 1.1.36. 

https://drive.google.com/open?id=0B9u6OGMG7xHcLVBTMU5iQ3ZyU0U

It should do what you asked for - matching RegExps on document level not on spelled lexeme level (bit st the moment it does still do the later). 

So once you put expression you show in first post it `ignoreRegExps` it starts to ignore `\begin{equation}(...)\end{equation}` sections.

Please put it to the test.

**Comment by innerlee** [comment]
It's working. Here is the testing regex
```json
"spellright.ignoreRegExps": [
        "/\\\\\\\\begin\\\\{equation\\\\}[\\\\s\\\\S]*?\\\\\\\\end\\\\{equation\\\\}/g",
        "/\\\\\\\\begin\\\\{equation\\\\\\\\*\\\\}[\\\\s\\\\S]*?\\\\\\\\end\\\\{equation\\\\\\\\*\\\\}/g",
        "/\\\\\\\\begin\\\\{framed\\\\}[\\\\s\\\\S]*?\\\\\\\\end\\\\{framed\\\\}/g",
        "/\\\\{bf\\\\}.*?end\\\\{bf\\\\}/g"
    ]
````
and the testing latex
```latex
\begin{framed}
     \missref xxabfd fdd
     dsfasdf
\end{framed}
\begin{equation}
    \#\Fix(f) = \sum(-1)^i b_i(M). fddfd
\end{equation}
\begin{equation*}
    \#\Fix(f) = \sum(-1)^i b_i(M). fddfd
\end{equation*}
{bf} fdd end{bf}
```
No error shown. Cool!

**Comment by bartosz-antosik** [comment]
Great. Will be included in next official release in few days. I am still unsure whether these RegExps should operate on two levels (on raw document as in this thread was requested and on separated words) but I will eventually decide whether to test on the lower level too.

**Comment by bartosz-antosik** [comment]
It has been published in version 1.1.38.
